src/docs/sdk/performance/ui-event-transactions.mdx

```diff
@@ -0,0 +1,173 @@
+the transaction when the timeout succeeds.
+* __Wait-for-children__: A transaction waits for all its child spans to finish before finishing itself.
+
+Before diving into the specification with all the edge cases, let's look at two simple examples:
+
+1. The user clicks a button that triggers some requests to the backend and stores data in the
+local database. The SDK can create a meaningful transaction in that case.
+2. The user clicks a button that solely validates some form data and triggers nothing that the
+SDK can automatically instrument. The SDK could only create a transaction without any spans, which would be valueless.
+
+# Specification
+
+Users can change the `idleTimeout` via the SDK config options. The default is 3.0 seconds. 
+
+The specification is written in the [Gherkin syntax](https://cucumber.io/docs/gherkin/reference/).
+
+**1. Starting UI Event transactions** 
+```
+Given an instrumentable UI event
+Then the SDK starts a UI event transaction 
+And starts the idle timeout with idleTimeout of the options
+```
+
+**5. Wait for children when starting span**
+```
+Given a UI event transaction
+When the SDK starts a child span
+Then the SDK cancels the idle timeout
+And waits for the child to finish
+```
+
+**4. Start timeout when the last span finishes**
+```
+Given a UI event transaction
+And the transaction has one or multiple running child spans
+And the transaction is waiting for its children to finish
+When the SDK finishes the last child span
+Then the SDK starts the idle timeout
+```
+
+**3. Don't reset timeout when the second last span finishes**
+```
+Given a UI event transaction
+And the transaction has two running child spans
+When the SDK finishes the first child span
+Then the SDK doesn't reset the idle timeout
+And waits for the children to finish
+```
+
+**2. Discard UI event transactions without child spans**
+```
+Given a UI event transaction
+And the transaction has no child spans
+When the idleTimeout times out
+Then the SDK discards the transaction 
+```
+
+**3. Set time to last finished child span**
+```
+Given a UI event transaction
+And the transaction has one finished child span
+And the transaction has one running child span
+When the running child span finishes
+Then the SDK finishes the transaction
+And trims the end time of the transaction to the one of the last finished child span
+```
+
+**6. Same UI element with same event**
+```
+Given an ongoing UI event transaction
+When the user triggers the same UI event with the same type for the same UI element
+Then the SDK resets the timeout
+And doesn't create a new transaction
+```
+
+**10. Same UI element with different event**
+If your SDK binds transactions to the scope see 11.
+```
+Given an ongoing UI event transaction
+When the user triggers the same UI element with a different event
+Or the user triggers a different UI element
+Then the SDK finishes the ongoing transaction
+And sets the status to OK
```

Comment:
I know this is already implemented, but it seems like the status could be `canceled` or something?
Also, the status is `ok` only if the user has not set a status manually.

Reply:
On Cocoa, we have `ok`, but we can change it.
```Swift
func test_DifferentUIElement_DifferentAction_FinishesTransaction() {
    let view1 = fixture.button
    callExecuteAction(action: "otherAction", target: fixture.target, sender: view1, event: TestUIEvent())
    
    let firstTransaction = try! XCTUnwrap(SentrySDK.span as? SentryTracer)
    
    let view2 = UIButton()
    callExecuteAction(action: action, target: fixture.target, sender: view2, event: TestUIEvent())
    
    assertFinishesTransaction(firstTransaction, operationClick)
}

private func assertFinishesTransaction(_ transaction: SentryTracer, _ operation: String) {
    XCTAssertTrue(transaction.isFinished)
    XCTAssertEqual(.ok, transaction.context.status)
    assertTransaction(name: "SentryTests.FirstViewController.\(expectedAction)", operation: operation)
    
    let transactions = getInternalTransactions()
    XCTAssertEqual(1, transactions.count)
}
```

https://github.com/getsentry/sentry-cocoa/blob/944bebeafd548a2cb391f8e6b2047ed37eec3755/Tests/SentryTests/Integrations/UIEvents/SentryUIEventTrackerTests.swift#L186-L196

https://github.com/getsentry/sentry-cocoa/blob/944bebeafd548a2cb391f8e6b2047ed37eec3755/Tests/SentryTests/Integrations/UIEvents/SentryUIEventTrackerTests.swift#L274

Reply:
On Android, if I understood it correctly, it forces to `ok`.
@romtsn to confirm.

Reply:
yes we use `ok`. I'm not sure about `canceled` because it doesn't sound like the user is cancelling/aborting their action - e.g. if they scrolled through twitter feed and found some interesting tweet and clicked to expand the thread - is it really cancelling the scrolling gesture, or just finishing the gesture logically, because there was an interesting thing requiring another gesture type?

`canceled` would make sense to me, if e.g. user clicked a button which triggered some network request, and this request was very slow to display the content, and the user pressed back/home button because they don't wanna wait anymore. Then it's for sure `canceled`, but we don't track system buttons yet.

Reply:
If the agreement was `ok`, let's keep it as it is, but Android should only set a status if there's no status, right now its overwriting the user's status in case it's set, @romtsn mind filling a bug for this?

Reply:
Btw just a small info, status `canceled` does not degrade the failure rate of transactions.
https://docs.sentry.io/product/performance/metrics/#failure-rate

Reply:
https://github.com/getsentry/sentry-java/issues/2378 filed